### PR TITLE
Add 3D attention_mask input support

### PR DESCRIPTION
**Pull request by SmileTM**
Add 3D attention_mask input support. (The PyTorch version support, but TF cant.)
Sometimes we need to input custom 3D attention_mask.

```python
import tensorflow as tf
from transformers import TFBertModel
plm_name = "bert-base-chinese"
plm = TFBertModel.from_pretrained(plm_name, return_dict=False)
i = tf.ones((1, 512), dtype=tf.int32)
m = tf.ones((1, 512, 512), dtype=tf.int32)
a, b = plm((i, m, i))
print(a)
print(b)
```

# What does this PR do?

Fixes # (issue) Add 3D attention_mask input support. 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?
Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

 
@LysandreJik @sgugger @patil-suraj

### CODE REVIEW

File: src/transformers/models/bert/modeling_tf_bert.py [hunk 1]
```diff
@@ -818,9 +823,17 @@ def call(
...
+            if len(attention_mask_shape) == 2:
+                extended_attention_mask = tf.reshape(
+                    inputs["attention_mask"], (attention_mask_shape[0], 1, 1, attention_mask_shape[1])
+                )
+            elif len(attention_mask_shape) == 3:
+                extended_attention_mask = tf.expand_dims(inputs["attention_mask"], 1)
+            else:
+                raise ValueError(
+                    f"Wrong shape for input_ids (shape {input_ids.shape}) or attention_mask (shape {attention_mask_shape})"
+                )
```
Comment (sgugger) [inline src/transformers/models/bert/modeling_tf_bert.py:835]: I think we can unindent all of this and change the first if in `elif`

**Review by sgugger — APPROVED** [review]
Thanks for adding support to this!

**Review by gante — COMMENTED** [review]
LGTM, in terms of the change itself. However, it is missing tests -- I'd recommend adding one, perhaps [here](https://github.com/huggingface/transformers/blob/master/tests/test_modeling_tf_bert.py#L397)

**Comment by Rocketknight1** [comment]
This method will be copied to RemBERT and RoBERTa as well, because those inherit the same `call()` method. I think we can accept the PR with just a BERT test, though I'm not sure - @sgugger WDYT?

**Comment by LysandreJik** [comment]
I think it's important to also have parity for the rest of the TF models - can all models accept the 3D attention mask? How hard is it to propagate this to all models? I don't think it's best if BERT can accept differing inputs than other TF models.
